src/ptvsd/common/launcher.py

```diff
@@ -78,5 +80,21 @@ def parse(argv):
     return it
 
 
+def _send_pid(pid):
+    if "PTVSD_PID_SERVER_PORT" not in os.environ:
+        return
+
+    from ptvsd.common import socket
+    port = int(os.environ["PTVSD_PID_SERVER_PORT"])
+    with socket.create_client() as sock:
+        sock.connect(("127.0.0.1", port))
+        sock.sendall(str(pid).encode("ascii"))
+
+
 if __name__ == "__main__":
+    # This is so we can correctly import socket from ptvsd.common
+    if "ptvsd" not in sys.modules:
+        sys.path[0] = os.path.dirname(__file__) + "/../../"
+        __import__("ptvsd")
+        del sys.path[0]
```

Comment:
Since it's a client socket, and it doesn't really need any special `setsockopt` sauce, it might be easier to just use the standard `socket` module directly?

Reply:
I tried this, seems that importing stdlib socket here conflicts with ptvsd.common.socket. Since that ptvsd's socket is in the current working directory.